src/pages/HomePage/components/CommentItem.jsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from "react";
+
+// components
+import UserCycle from "../../../components/UserCycle";
+
+// api
+import {
+    changeCommentData,
+    deleteComment,
+    getContact,
+} from "../../../utilities/api";
+import InputElement from "../../../components/InputElement";
+
+const CommentItem = ({ comment, getAllComments }) => {
+    const [commentUser, setCommentUser] = useState({});
+    const [isEdit, setIsEdit] = useState(false);
+    const [commentContent, setCommentContent] = useState("");
+
+    useEffect(() => {
+        getContact(comment.contactId).then((data) => setCommentUser(data));
+        setCommentContent(comment.content);
+        setIsEdit(false);
+    }, [comment]);
+
+    // functions
+
+    const submitDeleteComment = () => {
+        deleteComment(comment.postId, comment.id).then(() => getAllComments());
+    };
+
+    const submitChange = () => {
+        changeCommentData(comment.postId, comment.id, {
+            postId: comment.postId,
+            content: commentContent,
+            contactId: commentUser.id,
+        }).then(() => getAllComments());
```

Comment:
In cases like this, you can just give the .then a function to run instead of wrapping it in another function:

`.then(getAllComments)`